src/components/molecules/BaseCard/BaseCard.tsx

```diff
@@ -0,0 +1,96 @@
+import React, {useState} from "react";
+import { createUseStyles } from "react-jss";
+import { BaseProps } from '../../../models/BaseProps';
+import { getContainerOps } from "../../../utils/component-utils";
+
+export interface CardProps extends BaseProps {
+  frontFace?: JSX.Element,
+  backFace?: JSX.Element
+}
+
+export interface CardState {
+    visibleFace: 'front' | 'back'
+}
+
+const useStyles = createUseStyles({
+    // Generic classes
+  cardDimensions: {
+    width: 320,
+	  height: 240,
+    margin: '1em auto'
+  },
+  cardFace: {
+    extend: 'cardDimensions',
+    backfaceVisibility: 'hidden',
+	position: 'absolute',
+	top: 0,
+	left: 0,
+    color: 'cornSilk',
+    textAlign: 'center',
+    font: `3em/240px 'Helvetica Neue', Helvetica, sans-serif`,
+    boxShadow: '-5px 5px 5px #aaa',
+    transition: '0.6s'
+  },
+  // Modal classes
+  backVisible: {},
+  frontVisible: {},
+  // Style classes
+  card: {
+    extend: 'cardDimensions',
+    perspective: 1000,
+    cursor: 'pointer',
+    '&$backVisible $cardFlipper': {
+        transform: 'rotateY(180deg)'
+    },
+    '&$backVisible $cardFront': {
+      opacity: 0
+    },
+    '&$backVisible $cardBack': {
+      opacity: 1
+    },
+    '&$frontVisible $cardFront': {
+      opacity: 1
+    },
+    '&$frontVisible $cardBack': {
+      opacity: 0
+    }
+  },
+  cardFlipper: {
+    transition: '0.6s',
+	transformStyle: 'preserve-3d',
+	position: 'relative'
+  },
+  cardFront: {
+    extend: 'cardFace',
+    zIndex: 2,
+	  transform: 'rotateY(0deg)',
+    background: 'cornflowerBlue'
+  },
+  cardBack: {
+    extend: 'cardFace',
+    transform: 'rotateY(180deg)',
+    background: 'crimson'
+  }
+});
+
+/**
+ * Base card molecule
+ * @param props Card props
+ */
+export const BaseCard = (props: CardProps = {}) => {
+  const classes = useStyles();
+  const [cardState, setCardState] = useState<CardState>({visibleFace: 'front'});
```

Comment:
Here I think it could be simpler, we could represent the state of a card with a boolean, like:

```
const [isFrontVisible, setIsFrontVisible] = useState(false);
```

I don't realy like having a string describing which face is visible

Reply:
Now here is indeed a part where I couldn't make my mind.
Should we have several `useState` or a single one  holding the whole component? I am ok having it several, but I wasn't sure what is the pros / cons of this.

Reply:
Now, for the second point.
This is not a simple string, it's a string type. Compiler will force either `front` or `back`. Anything different, it will break.

I guess this makes the code more readable (`if visibleFace === 'back'` instead of `is isFrontVisible`)

Any arguments on why you don't like the string as describing which face is visible?

Reply:
In the end, after moving to `classnames`, having the boolean is  easier. I have changed